src/components/_shared/TabBar/TabBar.tsx

```diff
@@ -1,36 +1,54 @@
-import React from 'react'
-import PropTypes from 'prop-types'
+import React, { useRef } from 'react'
+import { uniqueId } from 'lodash'
+import * as colors from 'global_styles/colors'
 import styles from './TabBar.module.sass'
 
+function useId(): string {
+  const { current: id } = useRef(uniqueId())
+  return id
+}
+
+interface Props<T> {
+  title: string
+  className?: string
+  /** An array of objects like this: { value: 0, display: 'tab1' } */
+  tabs: T[]
+  selectedTab: T
+  /** takes the tab value and should return its label
+   *  (anything that can be renderd by react)
+   */
+  format?: (tab: T) => React.ReactNode
+  /** takes the tab value and should return its background color */
+  color?: (tab: T) => string
+  /** select handler */
+  onChange?: (tab: T) => void
+}
+
 /**
  * Renders selectable Tabs next to each other
  */
-const TabBar = props => {
-  const {
-    id,
-    tabs,
-    title,
-    selectedTab,
-    format,
-    color,
-    onChange,
-    className
-  } = props
+function TabBar<T>({
```

Comment:
Same thing here. The `TabBar` always expects the same `Tab`, not some kind of generic `T` that could look like anything.